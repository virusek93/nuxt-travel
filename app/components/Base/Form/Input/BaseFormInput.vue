<script lang="ts" setup>
import type { HtmlInputType } from '~~/types/general'

const props = withDefaults(
  defineProps<{
    name: string
    label: string
    required?: boolean
    type?: HtmlInputType
    placeholder?: string
    min?: string
    max?: string
    value?: string
    customClass?: string
  }>(),
  { type: 'text' },
)
const { value: fieldValue, errorMessage } = useField(() => props.name)
if (props.type === 'date' && props.value) {
  fieldValue.value = props.value
}
</script>

<template>
  <BaseFormField
    :name="name"
    :label="label"
    :error-message="errorMessage"
    :required="required"
  >
    <input
      :id="name"
      v-model="fieldValue"
      :name="name"
      :min="min"
      :max="max"
      :placeholder="placeholder"
      :type="type"
      :required="required"
      class="min-h-10 min-w-52 rounded-md border border-gray-300 px-4"
      :class="customClass"
    >
  </BaseFormField>
</template>
