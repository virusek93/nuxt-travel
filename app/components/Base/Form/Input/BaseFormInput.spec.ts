import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import BaseFormInput from './BaseFormInput.vue'

const props: InstanceType<typeof BaseFormInput>['$props'] = {
  name: 'name',
  label: 'label',
  type: 'text',
}

describe('BaseFormInput', () => {
  it('Should render and match the snapshot', () => {
    const wrapper = mount(BaseFormInput, { props })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Should pass props to input', () => {
    const optionalProps = {
      min: '10',
      max: '20',
      placeholder: 'test',
      required: true,
    }
    const wrapper = mount(BaseFormInput, {
      props: { ...props, ...optionalProps },
    })
    const input = wrapper.find('input')
    expect(input.attributes('min')).toBe(optionalProps.min)
    expect(input.attributes('max')).toBe(optionalProps.max)
    expect(input.attributes('placeholder')).toBe(optionalProps.placeholder)
    expect(input.attributes('required')).toBe('')
  })
})
