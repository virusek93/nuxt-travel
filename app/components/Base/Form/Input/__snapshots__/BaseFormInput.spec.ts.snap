// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseFormInput > Should render and match the snapshot 1`] = `
<div>
  <label
    class="flex cursor-pointer flex-col items-start gap-0.5 text-sm font-medium text-gray-500"
    data-test="form-label"
    for="name"
  >
    <span
      class="after:ml-0.5 after:hidden after:text-red-500 after:content-['*']"
    >
      label
    </span>
  </label>
  
  <input
    class="min-h-10 min-w-52 rounded-md border border-gray-300 px-4"
    id="name"
    name="name"
    type="text"
  />
  
  <!--v-if-->
</div>
`;
